uninstall.php

```diff
@@ -0,0 +1,173 @@
+					"DELETE FROM $wpdb->postmeta WHERE post_id IN ($ids_string)"
+				)
+			);
+
+			// Delete subscription posts.
+			$wpdb->query(
+				$wpdb->prepare(
+					"DELETE FROM $wpdb->posts WHERE ID IN ($ids_string)"
+				)
+			);
+
+			// Clear the post cache.
+			wp_cache_set_posts_last_changed();
+			wp_cache_delete_multiple( $subscription_post_ids, 'posts' );
+			wp_cache_delete_multiple( $subscription_post_ids, 'post_meta' );
+		}
+
+		// Delete relevant options from the options table.
+		delete_site_options();
+	}
+
+	/**
+	 * Delete all relevant options from the options table.
+	 */
+	function delete_site_options() {
+		global $wpdb;
+
+		$option_prefixes = array(
+			'dt_',
+			'_transient_dt_',
+			'_transient_timeout_dt_',
+		);
+
+		// Prepare the WHERE clause for the options table.
+		$where_clause = implode( ' OR ', array_fill( 0, count( $option_prefixes ), "option_name LIKE %s" ) );
+
+		$query = $wpdb->prepare(
+			sprintf(
+				"SELECT option_id FROM $wpdb->options WHERE %s;",
+				$where_clause
+			),
+			array_map( function( $prefix ) use ( $wpdb ) {
+				return $wpdb->esc_like( $prefix ) . '%';
+			}, $option_prefixes )
+		);
+
+		// Fetch the options to delete.
+		$options_to_delete = $wpdb->get_col( $query );
+
+		if ( ! empty( $options_to_delete ) ) {
+			$ids_string = implode( ',', array_map( 'intval', $options_to_delete ) );
+
+			// Delete the options using the retrieved IDs.
+			$wpdb->query(
+				$wpdb->prepare(
+					"DELETE FROM $wpdb->options WHERE option_id IN ($ids_string)"
+				)
+			);
+
+			// Flush the options cache.
+			wp_cache_delete_multiple( $options_to_delete, 'options' );
+
+			// Flush the alloptions cache.
+			wp_cache_delete( 'alloptions', 'options' );
+		}
+	}
+
+	/**
+	 * Delete all relevant options from the sitemeta table (multisite only).
+	 */
+	function delete_sitemeta_options() {
+		global $wpdb;
+
+		$option_prefixes = array(
+			'dt_',
+			'_site_transient_dt_',
+			'_site_transient_timeout_dt_',
+		);
+
+		// Prepare the WHERE clause for the sitemeta table.
+		$where_clause = implode( ' OR ', array_fill( 0, count( $option_prefixes ), "meta_key LIKE %s" ) );
+
+		$site_id = get_current_network_id();
+
+		$query = $wpdb->prepare(
+			sprintf(
+				"SELECT meta_id FROM $wpdb->sitemeta WHERE site_id = %%d AND (%s);",
+				$where_clause
+			),
+			array_merge( [ $site_id ], array_map( function( $prefix ) use ( $wpdb ) {
+				return $wpdb->esc_like( $prefix ) . '%';
+			}, $option_prefixes ) )
+		);
+
+		// Fetch the sitemeta to delete.
+		$sitemeta_to_delete = $wpdb->get_col( $query );
+
+		if ( ! empty( $sitemeta_to_delete ) ) {
+			$ids_string = implode( ',', array_map( 'intval', $sitemeta_to_delete ) );
+
+			// Delete the sitemeta using the retrieved IDs.
+			$wpdb->query(
+				$wpdb->prepare(
+					"DELETE FROM $wpdb->sitemeta WHERE meta_id IN ($ids_string)"
+				)
+			);
+
+			// Flush the site options cache.
+			wp_cache_delete_multiple( $sitemeta_to_delete, 'site-options' );
```

Comment:
This is using the option ID for the cache names, but it's cached using the pattern `$network_id:$option_name`

Reply:
This is using `meta_id` from the `sitemeta` table, do you mean we should pass `$network_id:{sitemeta.meta_key}`? Where we can use `sitemeta.site_id` as a `$network_id`. So something like this?:

`wp_cache_delete_multiple( [Array of {sitemeta.site_id:sitemeta.meta_key}], 'site-options' );`